src/pytom_tm/tmjob.py

```diff
@@ -92,6 +93,64 @@ def load_json_to_tmjob(
     return job
 
 
+def get_defocus_offsets(
+    patch_center_x: float,
+    patch_center_z: float,
+    tilt_angles: list[float, ...],
+    angles_in_degrees: bool = True,
+    invert_handedness: bool = False,
+) -> npt.NDArray[float]:
+    """Calculate the defocus offsets for a subvolume
+    based on the tilt geometry.
+
+    I used the definition from Pyle & Zianetti (https://doi.org/10.1042/BCJ20200715)
+    for the default setting of the defocus handedness. It assumes the defocus
+    increases for positive tilt angles on the right side of the sample (positive X
+    coordinate relative to the center).
+
+    The offset is calculated as follows:
+        z_offset = z_center * np.cos(tilt_angle) + x_center * np.sin(tilt_angle)
+
+    Parameters
+    ----------
+    patch_center_x: float
+        x center of subvolume relative to tomogram center
+    patch_center_z: float
+        z center of subvolume relative to tomogram center
+    tilt_angles: list[float, ...]
+        list of tilt angles
+    angles_in_degrees: bool, default True
+        whether tilt angles are in degrees or radians
+    invert_handedness: bool, default False
+        invert defocus handedness geometry
+
+    Returns
+    -------
+    z_offsets: npt.NDArray[float]
+        an array of defocus offsets for each tilt angle
+    """
+    n_tilts = len(tilt_angles)
+    x_centers = np.array(
+        [
+            patch_center_x,
+        ]
+        * n_tilts
+    )
+    z_centers = np.array(
+        [
+            patch_center_z,
+        ]
+        * n_tilts
+    )
```

Comment:
This feels surprisingly wastefull in lines, also probably quicker (and cleaner in my opinion) to build with [`np.full`](https://numpy.org/doc/stable/reference/generated/numpy.full.html)
```suggestion
    x_centers = np.full(n_tilts, patch_center_x)
    z_centers = np.full(n_tilts, patch_center_z)
```